pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -485,28 +486,24 @@ def _validate_monitor_key(self, trainer):
 
     def _get_metric_interpolated_filepath_name(
         self,
-        ckpt_name_metrics: Dict[str, Any],
+        monitor_candidates: Dict[str, Any],
         epoch: int,
         step: int,
         del_filepath: Optional[str] = None
     ) -> str:
-        filepath = self.format_checkpoint_name(epoch, step, ckpt_name_metrics)
-
-        version_cnt = 0
+        filepath = self.format_checkpoint_name(epoch, step, monitor_candidates)
+        version = self.STARTING_VERSION
         while self._fs.exists(filepath) and filepath != del_filepath:
-            filepath = self.format_checkpoint_name(epoch, step, ckpt_name_metrics, ver=version_cnt)
-            version_cnt += 1
-
+            filepath = self.format_checkpoint_name(epoch, step, monitor_candidates, ver=version)
+            version += 1
         return filepath
 
     def _monitor_candidates(self, trainer):
-        ckpt_name_metrics = deepcopy(trainer.logger_connector.logged_metrics)
-        ckpt_name_metrics.update(trainer.logger_connector.callback_metrics)
-        ckpt_name_metrics.update(trainer.logger_connector.progress_bar_metrics)
-        ckpt_name_metrics.update({"step": trainer.global_step, "epoch": trainer.current_epoch})
-        return ckpt_name_metrics
+        monitor_candidates = deepcopy(trainer.logger_connector.callback_metrics)
```

Comment:
```suggestion
        monitor_candidates = deepcopy(trainer.callback_metrics)
```
trainer.callback_metrics is a property